src/custom/state/claim/hooks/index.ts

```diff
@@ -275,16 +276,18 @@ export function useUserClaims(account: Account, optionalChainId?: SupportedChain
   return { claims: claimKey ? claimInfo[claimKey] : null, isLoading }
 }
 
-let fetch_deployment_timestamp_promise: Promise<number> | null = null
-function fetchDeploymentTimestamp(vCowContract: VCowType) {
-  if (!fetch_deployment_timestamp_promise) {
-    fetch_deployment_timestamp_promise = vCowContract.deploymentTimestamp().then((ts) => {
+const FETCH_DEPLOYMENT_TIME_PROMISES: Map<ChainId, Promise<number>> = new Map()
+function fetchDeploymentTimestamp(vCowContract: VCowType, chainId: ChainId): Promise<number> {
+  let deploymentTimePromise = FETCH_DEPLOYMENT_TIME_PROMISES.get(chainId)
+
+  if (!deploymentTimePromise) {
+    deploymentTimePromise = vCowContract.deploymentTimestamp().then((ts) => {
       console.log(`Deployment timestamp in seconds: ${ts.toString()}`)
```

Comment:
I don't think this works as intended https://jsfiddle.net/gac1kyfh/, I think you are not setting the chainId value on the map actually so its always going to be called

Reply:
ohh, true! i rushed this. Fix on its way!